src/backend/mpi/coherence.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file
+ * @brief This file implements selective coherence mechanisms for ArgoDSM
+ * @copyright Eta Scale AB. Licensed under the Eta Scale Open Source License. See the LICENSE file for details.
+ */
+
+#include "../backend.hpp"
+#include "swdsm.h"
+#include "write_buffer.hpp"
+
+// EXTERNAL VARIABLES FROM BACKEND
+extern control_data *cacheControl;
+extern unsigned long *globalSharers;
+extern pthread_mutex_t cachemutex;
+extern pthread_mutex_t wbmutex;
+extern sem_t ibsem;
+extern void * startAddr;
+extern char * barwindowsused;
+extern MPI_Win *globalDataWindow;
+extern MPI_Win sharerWindow;
+extern argo_statistics stats;
+extern argo_byte *touchedcache;
+extern MPI_Comm workcomm;
```

Comment:
Please document for all of these why they are needed. I think most/all of them could eventually be moved to proper API calls, but we need to write classes or APIs for them first. E.g. a cache class, a writebuffer class, and properly defined "backend" actions that deal with epochs in a defined way.

Reply:
These are now documented, all with `@deprecated` tags as they should not be needed long-term and replaced by appropriate API functions or carried out in other modules.